main/php_syslog.c

```diff
@@ -68,11 +68,19 @@ PHPAPI void php_syslog(int priority, const char *format, ...) /* {{{ */
 			break;
 		}
 
-		if (c != '\n')
+		/* check for NVT ASCII only unless test disabled */
+		if (((0x20 <= c) && (c <= 0x7e)) || (!PG(syslog_ascii_only) && (c >= 0x80)))
```

Comment:
Unless I miss something in the logic, I think that the `(c >= 0x80)` should be dropped as it changes the current behavior which allows control characters by default. The issue is that [RFC 5424 - section 6](https://tools.ietf.org/html/rfc5424#section-6) actually allows any character for MSG-ANY so this should be followed by default IMHO.

Reply:
If `syslog_ascii_only` is false, then any upper-half characters from any ISO-8859-_x_ will be printed.

We talked about this quite a bit as I remember, and the consensus was that there wasn't a compelling reason for logging to include non-printing characters, and indeed doing so might present a vulnerability.

Reply:
If `syslog_ascii_only` is false, then any upper-half characters from any ISO-8859-_x_ will be printed.

We talked about this quite a bit as I remember, and the consensus was that there wasn't a compelling reason for logging to include non-printing characters, and indeed doing so might present a vulnerability.

Reply:
I remember that one now. I think we actually ended up without any consensus... :) Will think about this later.